dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallback.java

```diff
@@ -358,15 +358,18 @@ public void initWithHosts(Map<Host, HostConnectionPool<CL>> hPools) {
 		// Initialize Local selector
 		Map<HostToken, HostConnectionPool<CL>> localPools = getHostPoolsForRack(tokenPoolMap, localRack);
 		localSelector.initWithHosts(localPools);
-		if (localSelector.isTokenAware() && localRack != null) {
-		   replicationFactor.set(calculateReplicationFactor(allHostTokens));
+		if (localSelector.isTokenAware()) {
+		   replicationFactor.set(calculateReplicationFactorForDC(allHostTokens, cpConfig.getLocalDataCenter()));
 		}
 
 		// Initialize Remote selectors
 		Set<String> remoteRacks = new HashSet<String>();
 		for (Host host : hPools.keySet()) {
 			String rack = host.getRack();
-			if (localRack != null && !localRack.isEmpty() && rack != null && !rack.isEmpty() && !localRack.equals(rack)) {
+			if(localRack == null && rack != null) {
+				remoteRacks.add(rack);
+			}
+			else if (localRack != null && !localRack.isEmpty() && rack != null && !rack.isEmpty() && !localRack.equals(rack)) {
```

Comment:
Isn't the same as just removing the 'localrack != null' check in L372? That way we get the same behavior in one if condition.